lib/Model/TypeSystemPrinter.cpp

```diff
@@ -591,6 +592,64 @@ void TypeSystemPrinter::print(const model::DynamicFunction &F) {
   }
 }
 
+void TypeSystemPrinter::dumpSegmentNode(const model::Segment &S, int NodeID) {
+  const model::Type *T = S.Type().empty() ? nullptr : S.Type().getConst();
+
+  // Print the name of the node
+  Out << "node_" << to_string(NodeID) << "[";
+
+  // Choose the node's border color
+  auto Color = Orange;
```

Comment:
[B.15](https://pad.rev.ng/s/SJNUet_ku#). Below too.

Reply:
Right. Looking around I've seen that this file was pretty `auto`-heavy, and I've converted a bunch of them to using explicit types.